src/ceph-disk/tests/ceph-disk.sh

```diff
@@ -381,17 +421,24 @@ function run() {
     default_actions+="test_activate_dir_magic "
     default_actions+="test_activate_dir "
     default_actions+="test_keyring_path "
-    default_actions+="test_mark_init "
+    [ `uname` != FreeBSD ] && default_actions+="test_mark_init "
     default_actions+="test_zap "
-    default_actions+="test_activate_dir_bluestore "
+    [ `uname` != FreeBSD ] && default_actions+="test_activate_dir_bluestore "
     default_actions+="test_ceph_osd_mkfs "
     local actions=${@:-$default_actions}
     for action in $actions  ; do
         setup $dir || return 1
         set -x
-        $action $dir || return 1
+        $action $dir 
         set +x
-        teardown $dir || return 1
+	status=$?
+	if [ $status -ne 0 ] ; then
+	    # Keep data for postmortum analysis
+            teardown_no_rm $dir 
```

Comment:
maybe we can have this change for another PR? it don't think this blocks the FreeBSD porting.